### PR TITLE
Allow submission of execution length in Enforcer register and dispute

### PR DESCRIPTION
**Pull request by peara**

### CODE REVIEW

**Comment by johannbarbie** [comment]
@peara thanks for this.
can you help me understand where the step that you save will be used?

**Comment by peara** [comment]
@johannbarbie 
the submitted step is used in the `respond` procedure to decide the next required submission.
also when the different is found, the detail execution of the found step is required.

File: contracts/Enforcer.sol [hunk 1]
```diff
@@ -37,25 +38,27 @@ contract Enforcer {
         bondAmount = _bondAmount;
     }
 
-    // register a new execution 
-    function register(bytes memory _code, bytes memory _callData, bytes32 _endHash) public payable returns(bytes32 executionId) {
+    // register a new execution
+    function register(bytes memory _code, bytes memory _callData, bytes32 _endHash, uint256 _step)
```
Comment (johannbarbie) [inline contracts/Enforcer.sol]: i suggest to rename `_step` to `_steps` or `_executionLength`.

File: contracts/Enforcer.sol [hunk 2]
```diff
@@ -37,25 +38,27 @@ contract Enforcer {
...
+        executions[executionId] = Execution(block.number, _endHash, _step, msg.sender);
         bonds[msg.sender] += bondAmount;
         emit Registered(executionId, msg.sender, _code, _callData);
     }
 
     // starts a new dispute
-    function dispute(bytes32 _executionId, bytes32 _endHash) public payable {
-        require(executions[_executionId].startBlock != 0);
-        require(executions[_executionId].startBlock + challengePeriod > block.number);
+    function dispute(bytes32 _executionId, bytes32 _endHash, uint256 _step) public payable {
```
Comment (johannbarbie) [inline contracts/Enforcer.sol]: same, suggest renaming to `_steps` or `_executionLength`

File: contracts/mocks/VerifierMock.sol [hunk 3]
```diff
@@ -6,50 +6,56 @@ import "../IVerifier.sol";
...
+        _;
+    }
+
+    function initGame(bytes32 _executionId, bytes32 _solverHash, uint256 _solverStep,
+        bytes32 _challengerHash, uint256 _challengerStep, address _challenger) public onlyEnforcer() {
+        bytes32 disputeId = keccak256(abi.encodePacked(_executionId, _challenger));
+        require(disputes[disputeId].lastQueryBlock == 0);
+        emit DisputeStart(disputeId);
+        disputes[disputeId] = Dispute(block.number, _challenger, _executionId);
+        _solverHash;
```
Comment (johannbarbie) [inline contracts/mocks/VerifierMock.sol]: what is this good for?

Comment (peara) [reply contracts/mocks/VerifierMock.sol]: it's to silence solhint in those mock function which doesn't do anything
also my habit for this particular err...

Comment (johannbarbie) [reply contracts/mocks/VerifierMock.sol]: i see.
i guess you know, but you can just remove the name but leave the type, like this:
`function(address owner)` -> `function(address)`
also silences the error

**Pull request merged**